iroha/test_network/tests/mock.rs

```diff
@@ -0,0 +1,790 @@
+                drop(self.round(txs).await);
+            }
+        }
+
+        #[async_trait::async_trait]
+        impl<Q, G, W> Handler<Round> for Sumeragi<Q, G, W>
+        where
+            Q: QueueTrait,
+            G: GenesisNetworkTrait,
+            W: WorldTrait,
+        {
+            type Result = ();
+            async fn handle(&mut self, Round(txs): Round) -> Self::Result {
+                drop(self.round(txs).await);
+            }
+        }
+    }
+
+    pub mod world {
+        use std::ops::{Deref, DerefMut};
+
+        use iroha::prelude::*;
+        use iroha::{tx::Domain, wsv::WorldTrait};
+        use iroha_data_model::prelude::*;
+        use iroha_data_model::world::World;
+        use once_cell::sync::Lazy;
+
+        #[derive(Debug, Clone, Default)]
+        pub struct WithRoot(World);
+
+        impl Deref for WithRoot {
+            type Target = World;
+            fn deref(&self) -> &World {
+                &self.0
+            }
+        }
+
+        impl DerefMut for WithRoot {
+            fn deref_mut(&mut self) -> &mut Self::Target {
+                &mut self.0
+            }
+        }
+
+        pub static ROOT_KEYS: Lazy<KeyPair> = Lazy::new(|| KeyPair::generate().unwrap());
+        pub static ROOT_ID: Lazy<AccountId> = Lazy::new(|| AccountId::new("root", "global"));
+        pub static ROOT: Lazy<Account> = Lazy::new(|| {
+            let mut account = Account::new(ROOT_ID.clone());
+            account.signatories.push(ROOT_KEYS.public_key.clone());
+            account
+        });
+        pub static GLOBAL: Lazy<Domain> = Lazy::new(|| {
+            let mut domain = Domain::new("global");
+            drop(domain.accounts.insert(ROOT_ID.clone(), ROOT.clone()));
+            domain
+        });
+
+        impl WorldTrait for WithRoot {
+            /// Creates `World` with these `domains` and `trusted_peers_ids`
+            fn with(
+                domains: impl IntoIterator<Item = (Name, Domain)>,
+                trusted_peers_ids: impl IntoIterator<Item = PeerId>,
+            ) -> Self {
+                Self(iroha_data_model::world::World::with(
+                    vec![(GLOBAL.name.clone(), GLOBAL.clone())]
+                        .into_iter()
+                        .chain(domains),
+                    trusted_peers_ids,
+                ))
+            }
+        }
+
+        pub fn sign_tx(isi: impl IntoIterator<Item = Instruction>) -> VersionedAcceptedTransaction {
+            let tx = Transaction::new(isi.into_iter().collect(), ROOT_ID.clone(), 100_000)
+                .sign(&ROOT_KEYS)
+                .unwrap();
+            let tx = VersionedAcceptedTransaction::from_transaction(tx, 4096).unwrap();
+            dbg!(tx.hash());
+            tx
+        }
+    }
+}
+
+/// Checks if `blocks` number of blocks is applied on peer.
+async fn blocks_applied_peer(channel: &mut mpsc::Receiver<Stored>, blocks: usize) -> usize {
```

Comment:
The naming is very confusing and comment does not help. The function below has the same comment but does a very different thing.